src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,128 @@
+            } catch (IOException e) {
+                throw new DukeException("Unable to create file...");
+            }
+        }
+
+    }
+
+    /**
+     * Returns an array list containing tasks from previous session by extracting details
+     * of tasks from local file and creating corresponding task objects.
+     *
+     * @return An array list containing tasks from previous session.
+     * @throws DukeException  If there is no file available to load.
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> list = new ArrayList<>();
+        try {
+            Scanner scanner = new Scanner(this.file);
+            while (scanner.hasNextLine()) {
+                String[] currLine = scanner.nextLine().split(" \\| ");
+                String typeOfTask = currLine[0];
+                switch (typeOfTask) {
+                case "T":
+                    Task todo = new Todo(currLine[2]);
+                    if (Integer.parseInt(currLine[1]) == 1) {
+                        todo.doTask();
+                    }
+                    list.add(todo);
+                    break;
+                case "D":
+                    Task deadline = new Deadline(currLine[2], currLine[3]);
+                    if (Integer.parseInt(currLine[1]) == 1) {
+                        deadline.doTask();
+                    }
+                    list.add(deadline);
+                    break;
+                case "E":
+                    Task event = new Event(currLine[2], currLine[3]);
+                    if (Integer.parseInt(currLine[1]) == 1) {
+                        event.doTask();
+                    }
+                    list.add(event);
+                    break;
+                }
+            }
+            return list;
+        } catch (FileNotFoundException e) {
+            throw new DukeException("Unable to load file.. :c");
+        }
+    }
+
+    /**
+     * Saves the tasks in the hard disk when the task list changes.
+     *
+     * @param taskToAdd A string representing details of the task to be added to the file.
+     * @throws DukeException  If unable to edit file.
+     */
+    public void saveTaskList(String taskToAdd) throws DukeException {
+        try {
+            FileWriter fw = new FileWriter(this.file, true);
+            fw.write(taskToAdd + "\n");
+            fw.close();
+        } catch (IOException e) {
+            throw new DukeException("Error editing file!");
+        }
+    }
+
+    /**
+     * Edit the existing tasks in the hard disk when the task list changes.
+     *
+     * @param newTask A string representing new details of the task already present in the file.
+     * @param taskNum Position of task in file.
+     * @param delete If the action is to delete the task in file.
+     * @throws DukeException  If unable to edit file.
+     */
+    public void editTaskList(String newTask, int taskNum, boolean delete) throws DukeException {
```

Comment:
I think you could use a clearer name for boolean so that we know it is a boolean value such as isDelete or shouldDelete:)